src/irmin-pack/pack.ml

```diff
@@ -140,9 +144,6 @@ struct
     let lock = Lwt_mutex.create () in
     let dict = Dict.v ~fresh ~readonly root in
     let block = IO.v ~fresh ~version:current_version ~readonly file in
-    if IO.version block <> current_version then
-      Fmt.failwith "invalid version: got %S, expecting %S" (IO.version block)
-        current_version;
```

Comment:
This check seems like it might still be worthwhile.

Reply:
I use `v` from irmin_pack and thus `unsafe_v` from pack in the migrate function, so this needs to accept both V1 and V2. there is a check that files are in either V1 or V2 in IO (https://github.com/mirage/irmin/pull/1015/files#diff-8a044b6b9c9b0a69a5f770d759c9d1b9R264)